pkg/polaris/ctl_polaris.go

```diff
@@ -67,317 +66,264 @@ type CRSpecBuilderFromCobraFlags struct {
+		cmd.Flags().StringVar(&ctl.flagTree.PostgresSize, "postgres-size", POSTGRES_PV_SIZE, "Persistent volume claim size to use for postgres. Only applicable if --enable-postgres-container is set to true")
+		cmd.Flags().StringVar(&ctl.flagTree.StorageClass, "storage-class", ctl.flagTree.StorageClass, "Set the storage class to use for all persistent volume claims\n")
 	}
 
 	// reporting related flags
-	cmd.Flags().BoolVar(&ctl.EnableReporting, "enable-reporting", GetPolarisDefault().EnableReporting, "Enable Reporting Platform")
-	cmd.Flags().StringVar(&ctl.ReportStorageSize, "reportstorage-size", GetPolarisDefault().ReportingSpec.ReportStorageDetails.Storage.StorageSize, "Persistent volume claim size for reportstorage. Only applicable if --enable-reporting is set to true")
-
-	// flags that are add-ons (helpful, but not required to set up an environment
-	cmd.Flags().StringVar(&ctl.Registry, "registry", ctl.Registry, "Docker registry e.g. docker.io/myuser")
-	cmd.Flags().StringVar(&ctl.ImagePullSecrets, "pull-secret", ctl.ImagePullSecrets, "Pull secret when using a private registry\n")
+	cmd.Flags().BoolVar(&ctl.flagTree.EnableReporting, "enable-reporting", false, "Enable Reporting Platform")
+	cmd.Flags().StringVar(&ctl.flagTree.ReportStorageSize, "reportstorage-size", REPORT_STORAGE_PV_SIZE, "Persistent volume claim size for reportstorage. Only applicable if --enable-reporting is set to true")
 
 	cmd.Flags().SortFlags = false
 }
 
-// CheckValuesFromFlags returns an error if a value stored in the struct will not be able to be
-// used in the spec
-func (ctl *CRSpecBuilderFromCobraFlags) CheckValuesFromFlags(flagset *pflag.FlagSet) error {
+// CheckValuesFromFlags returns an error if a value set by a flag is invalid
+func (ctl *HelmValuesFromCobraFlags) CheckValuesFromFlags(flagset *pflag.FlagSet) error {
 	return nil
 }
 
-// GenerateCRSpecFromFlags checks if a flag was changed and updates the spec with the value that's stored
-// in the corresponding struct field
-func (ctl *CRSpecBuilderFromCobraFlags) GenerateCRSpecFromFlags(flagset *pflag.FlagSet) (interface{}, error) {
+func validateEmail(email string) bool {
+	Re := regexp.MustCompile(`^[a-zA-Z0-9._%+\-]+@[a-z0-9.\-]+\.[a-z]+$`)
+	return Re.MatchString(email)
+}
+
+func validateFQDN(fqdn string) bool {
+	Re := regexp.MustCompile(`^[a-z0-9\-]+([.][a-z0-9\-]+)*\.[a-z]+$`)
+	return Re.MatchString(fqdn)
+}
+
+// GenerateHelmFlagsFromCobraFlags checks each flag in synopsysctl and updates the map to
+// contain the corresponding helm chart field and value
+func (ctl *HelmValuesFromCobraFlags) GenerateHelmFlagsFromCobraFlags(flagset *pflag.FlagSet) (map[string]interface{}, error) {
 	err := ctl.CheckValuesFromFlags(flagset)
 	if err != nil {
 		return nil, err
 	}
-	flagset.VisitAll(ctl.SetCRSpecFieldByFlag)
-	return ctl.spec, nil
+	ctl.args["global"] = map[string]interface{}{"environment": "onprem"}
```

Comment:
Is this {environment: onprem} needed for the Polaris Helm Chart? Or just Polaris-Reporting?

If it's just Polaris-Reporting then the Polaris Helm Chart could set it to onprem instead of doing it here right?

Reply:
Nevermind, I see that it exists in Values.yaml :)